src/extensions/rate_limit/mod.rs

```diff
@@ -27,16 +27,23 @@ pub struct RateLimitConfig {
 
 #[derive(Deserialize, Debug, Clone, Default)]
 pub struct Rule {
-    // burst is the maximum number of requests that can be made in a period
+    /// burst is the maximum number of requests that can be made in a period
     pub burst: u32,
-    // period is the period of time in which the burst is allowed
+    /// period is the period of time in which the burst is allowed
     #[serde(default = "default_period_secs")]
     pub period_secs: u64,
     // jitter_millis is the maximum amount of jitter to add to the rate limit
     // this is to prevent a thundering herd problem https://en.wikipedia.org/wiki/Thundering_herd_problem
     // e.g. if jitter_up_to_millis is 1000, then additional delay of random(0, 1000) milliseconds will be added
     #[serde(default = "default_jitter_up_to_millis")]
     pub jitter_up_to_millis: u64,
+    /// Return a rate limit jsonrpc error directly if true.
+    #[serde(default = "default_no_blocking")]
+    pub non_blocking: bool,
```

Comment:
how about use `blocking: bool` as option and it's false by default

Reply:
I can change this PR if you agree with it

Reply:
Both are ok.